src/main/java/com/bobocode/EntityPersister.java

```diff
@@ -0,0 +1,65 @@
+package com.bobocode;
+
+import com.bobocode.annotation.Column;
+import com.bobocode.exception.BibernateSQLException;
+import com.bobocode.exception.EntityMappingException;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.sql.DataSource;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@AllArgsConstructor
+public class EntityPersister {
+    private final DataSource dataSource;
+
+    public <T> List<T> select(Class<T> type, String query, List<Object> columnValuesToFilter) {
+        try (Connection connection = dataSource.getConnection()) {
+            try (PreparedStatement statement = connection.prepareStatement(query)) {
+                for (int i = 0; i < columnValuesToFilter.size(); i++) {
+                    statement.setObject(i + 1, columnValuesToFilter.get(i));
+                }
+                log.trace(statement.toString());
+
+                ResultSet resultSet = statement.executeQuery();
+                return processResultSet(type, resultSet);
+            }
+        } catch (SQLException e) {
+            throw new BibernateSQLException("Error loading data from DB", e);
+        }
+    }
+
+    @SuppressWarnings("java:S3011")
+    private <T> List<T> processResultSet(Class<T> type, ResultSet resultSet) {
+        List<T> resultList = new ArrayList<>();
+        try {
+            while (resultSet.next()) {
+                T obj = type.getConstructor().newInstance();
+                Field[] fields = type.getDeclaredFields();
+                for (Field field : fields) {
+                    field.setAccessible(true);
+                    String columnName = field.getDeclaredAnnotation(Column.class).value();
+                    field.set(obj, resultSet.getObject(columnName));
+                }
+                resultList.add(obj);
+            }
+        } catch (SQLException e) {
+            throw new BibernateSQLException("Error parsing data got from DB", e);
+        } catch (InstantiationException
+                 | IllegalAccessException
+                 | NoSuchMethodException
+                 | InvocationTargetException e) {
+            throw new EntityMappingException("Entity must have public zero-argument constructor!");
```

Comment:
Does this message represent all the exceptions in this list? Would it make sense to write some generic message like "Entity Mapping Error" and pass the trace of the exception that was thrown? I mean `throw new EntityMappingException("Entity mapping error", e);` 🤔

Reply:
Yes, it make sense.